### DISCUSSION TITLE
.NET Aspire Starter Application not working..

### DISCUSSION THREAD

**Discussion by newpost** [microsoft/aspire#4905] [General]
```
info: Aspire.Hosting.DistributedApplication[0]
      Aspire version: 8.0.0+d215c528c07c7919c3ac30b35d92f4e51a60523b
info: Aspire.Hosting.DistributedApplication[0]
      Distributed application starting.
info: Aspire.Hosting.DistributedApplication[0]
      Application host directory is: D:\03test\cs-demos\AspireApp1\AspireApp1.AppHost
fail: Aspire.Hosting.Dcp.dcpctrl.ServiceReconciler[0]
      could not start the proxy {"ServiceName": {"name":"cache"}, "Reconciliation": 4, "error": "cound not start the proxy for the service: listen tcp [::1]:61895: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted."}
fail: Aspire.Hosting.Dcp.dcpctrl.ServiceReconciler[0]
      could not start the proxy {"ServiceName": {"name":"apiservice"}, "Reconciliation": 5, "error": "cound not start the proxy for the service: listen tcp [::1]:5457: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted."}
fail: Aspire.Hosting.Dcp.dcpctrl.ServiceReconciler[0]
      could not start the proxy {"ServiceName": {"name":"webfrontend"}, "Reconciliation": 6, "error": "cound not start the proxy for the service: listen tcp [::1]:5279: bind: Only one usage of each socket address (protocol/network address/port) is normally permitted."}
fail: Microsoft.Extensions.Hosting.Internal.Host[11]
      Hosting failed to start
      Polly.Timeout.TimeoutRejectedException: The operation didn't complete within the allowed timeout of '00:00:10'.
       ---> System.Threading.Tasks.TaskCanceledException: A task was canceled.
         at k8s.Watcher`1.CreateWatchEventEnumerator(Func`1 streamReaderCreator, Action`1 onError, CancellationToken cancellationToken)+MoveNext()
         at k8s.Watcher`1.CreateWatchEventEnumerator(Func`1 streamReaderCreator, Action`1 onError, CancellationToken cancellationToken)+System.Threading.Tasks.Sources.IValueTaskSource<System.Boolean>.GetResult()
         at Aspire.Hosting.Dcp.KubernetesService.WatchAsync[T](String namespaceParameter, CancellationToken cancellationToken)+MoveNext() in /_/src/Aspire.Hosting/Dcp/KubernetesService.cs:line 209
         at Aspire.Hosting.Dcp.KubernetesService.WatchAsync[T](String namespaceParameter, CancellationToken cancellationToken)+MoveNext() in /_/src/Aspire.Hosting/Dcp/KubernetesService.cs:line 209
```

**Comment by davidfowl** [comment]
Can you share more about your environment:
- What OS you are using
- The output of dotnet --info
- What your launch settings for each project looks like:

cc @karolz-ms @danegsta

**Comment by s084088** [comment]
I have the same situation.
Just create a new project and no file has been modified

![image](https://github.com/dotnet/aspire/assets/42663215/bdc51b01-582d-41dd-8190-e6640b404d54)


`
C:\Users\s0840>dotnet --info
.NET SDK:
 Version:           8.0.300
 Commit:            326f6e68b2
 Workload version:  8.0.300-manifests.6751eab7
 MSBuild version:   17.10.4+10fbfbf2e

运行时环境:
 OS Name:     Windows
 OS Version:  10.0.22631
 OS Platform: Windows
 RID:         win-x64
 Base Path:   C:\Program Files\dotnet\sdk\8.0.300\

已安装 .NET 工作负载:
 [android]
   安装源文件: SDK 8.0.300, VS 17.10.34916.146
   清单版本:    34.0.95/8.0.100
   清单路径:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.android\34.0.95\WorkloadManifest.json
   安装类型:        FileBased

 [aspire]
   安装源文件: SDK 8.0.300
   清单版本:    8.0.1/8.0.100
   清单路径:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.aspire\8.0.1\WorkloadManifest.json
   安装类型:        FileBased

 [ios]
   安装源文件: SDK 8.0.300, VS 17.10.34916.146
   清单版本:    17.2.8053/8.0.100
   清单路径:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.ios\17.2.8053\WorkloadManifest.json
   安装类型:        FileBased

 [maccatalyst]
   安装源文件: SDK 8.0.300, VS 17.10.34916.146
   清单版本:    17.2.8053/8.0.100
   清单路径:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.maccatalyst\17.2.8053\WorkloadManifest.json
   安装类型:        FileBased

 [maui-windows]
   安装源文件: SDK 8.0.300, VS 17.10.34916.146
   清单版本:    8.0.21/8.0.100
   清单路径:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.maui\8.0.21\WorkloadManifest.json
   安装类型:        FileBased

 [tvos]
   安装源文件: SDK 8.0.300, VS 17.10.34916.146
   清单版本:    17.2.8053/8.0.100
   清单路径:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.sdk.tvos\17.2.8053\WorkloadManifest.json
   安装类型:        FileBased

 [wasm-tools]
   安装源文件: SDK 8.0.300, VS 17.10.34916.146
   清单版本:    8.0.5/8.0.100
   清单路径:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.workload.mono.toolchain.current\8.0.5\WorkloadManifest.json
   安装类型:        FileBased

 [wasm-tools-net6]
   安装源文件: SDK 8.0.300, VS 17.10.34916.146
   清单版本:    8.0.5/8.0.100
   清单路径:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.workload.mono.toolchain.net6\8.0.5\WorkloadManifest.json
   安装类型:        FileBased

 [wasm-tools-net7]
   安装源文件: SDK 8.0.300, VS 17.10.34916.146
   清单版本:    8.0.5/8.0.100
   清单路径:       C:\Program Files\dotnet\sdk-manifests\8.0.100\microsoft.net.workload.mono.toolchain.net7\8.0.5\WorkloadManifest.json
   安装类型:        FileBased


Host:
  Version:      8.0.5
  Architecture: x64
  Commit:       087e15321b

.NET SDKs installed:
  8.0.300 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.19 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 8.0.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.18 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.19 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 8.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 6.0.18 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.30 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.19 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 8.0.5 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download

`

**Comment by davidfowl** [comment]
@karolz-ms @dbreshears can we look into ths?

**Comment by danegsta** [comment]
Based on the error messages in these reports, there's a very good chance that there's already a second copy of the application running. `Only one usage of each socket address (protocol/network address/port) is normally permitted.` is an error that's thrown when a port is already bound on the system. If it were a single conflicting port, it might be a random conflict, but the fact that multiple ports are conflicting is what makes me think it's likely multiple copies of the same app.

If this is still reproducing, can you double check that you don't have any other copies of your application running in the background?

**Comment by s084088** [comment]
I'm certain that I only started one instance of the program, as I have restarted my computer. Moreover, even when I manually change to a random port, the error still persists. I suspect there might be some reason causing it to start twice.

**Comment by danegsta** [comment]
Considering that the error screenshot references IPv6 addresses, I'm suspicious this is an IPv6 networking issue (likely the last entry in your hosts file for localhost is [::1], but your IPv6 network either has some particular firewall settings that are preventing us from binding addresses.

We've made some improvements that'll ship in 8.1 that will allow overriding the IP type (IPv4 vs IPv6) that is preferred for binding service ports. Once 8.1 ships, you can set the `DCP_IP_VERSION_PREFERENCE` environment variable to `IPv4` or `IPv6` to prefer a particular IP type for service bindings.

Since there isn't a specific repro for this issue and it's potentially down to specific machine configurations, I'm going to convert this to a discussion. If we're able to narrow it down to a specific set of repro steps, we can convert it back to an issue.

**Reply by jcbeck37** [reply]
> We've made some improvements that'll ship in 8.1 that will allow overriding the IP type (IPv4 vs IPv6) that is preferred for binding service ports. Once 8.1 ships, you can set the `DCP_IP_VERSION_PREFERENCE` environment variable to `IPv4` or `IPv6` to prefer a particular IP type for service bindings.

_Where_ do you set this value?

**Reply by karolz-ms** [reply]
It is an *environment variable* (OS concept), so depends on how you start the application. E.g. if running the application from Visual Studio, you can use a launch settings file.

**Reply by jcbeck37** [reply]
> It is an _environment variable_ (OS concept), so depends on how you start the application. E.g. if running the application from Visual Studio, you can use a launch settings file.

Thank you! I ended up setting `NO_PROXY` (system environment variable) for my local development which was the real fix for my issue.

**Comment by a764578566** [comment]
I have the same situation.
![image](https://github.com/user-attachments/assets/7aa5ae45-a92a-48c0-a741-d9151a8d60b5)
![image](https://github.com/user-attachments/assets/266a586f-053d-4490-acd0-487488c5af6c)
![image](https://github.com/user-attachments/assets/b866bf29-872d-42cb-8e1b-57c679abdb5e)

**Reply by newpost** [reply]
How did you solve it? @a764578566

**Comment by markat1** [comment]
I have experienced this  - I often just reboot my computer, which makes it certain for me that the current instance goes away. 

Probably not the best solution, but that's how I do it at the moment.

Then I make sure not to spin up an instance of my aspire right after each other, which gives it time to remove the current spunned instance.

**Reply by davidfowl** [reply]
And by this you mean this exact error?

**Reply by markat1** [reply]
sure looks like the same error I experience - something like "Only one usage of each socket address (protocol/network address/port) is normally permitted"

**Reply by davidfowl** [reply]
If you could paste the error message in here that would be helpful

**Reply by newpost** [reply]
After updating to 24h2, the problem was automatically fixed, damn it!
